client/client-multi/client-c/listfiles/UnitTest.cmake

```diff
@@ -26,154 +26,203 @@ if(CUNIT_FOUND)
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DKAA_TEST_CUNIT_ENABLED")
 endif()
 
+################################################################################
+# Creates an unit test with given name and dependencies.
+#
+# TODO: move this function to the separate module, after new design will be implemented.
+#       When new SDK build system will be ready, tests will be encapsulated
+#       within each production module. Test system helpers, like this function
+#       will be moved to separate cmake build modules or scripts.
+#
+# Syntax:
+# kaa_add_unit_test(NAME test_name
+#                   SOURCES test_sources_files...
+#                   [DEPENDS list_of_dependencies...]
+#                   [INC_DIRS list_of_include_directories...])
```

Comment:
I'm afraid it cannot be done via [CMakeParseArguments](https://cmake.org/cmake/help/v3.0/module/CMakeParseArguments.html)

Most likely this will require implementation of iteration over list of argument that will increase code complexity.